### PR TITLE
Login and user creation

### PR DESCRIPTION
**Pull request by TimMikeladze**
Redis implementation of the accounts database interface.

Pending #1

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    const userId = shortid.generate();
+    const {
+      username,
+      password,
+      email,
+      profile,
+    } = user;
+    if (username) {
+      pipeline.set(`username:${username}`, userId);
+      pipeline.hmset(`user:${userId}`, { username });
```
Comment (grantga) [inline src/index.js]: Wouldn't this be better in two hashes like:
hmset("userByName", "username", userId)
hmset("userById", "id", username )

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    }
+    if (password) {
+      const hash = await encryption.hashPassword(password);
+      pipeline.set(`password:${userId}`, hash);
```
Comment (grantga) [inline src/index.js]: Again should this be in a hash:
hmset('passwordById', userId, hash)

File: src/index.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    }
+    if (email) {
+      pipeline.set(`email:${email}`, userId);
+      pipeline.sadd(`user:${userId}:emails`, email);
+    }
+    if (profile) {
+      pipeline.hmset(`user:${userId}`, { profile });
```
Comment (grantga) [inline src/index.js]: Is profile a Map--this then adds a key for each member of profile. Don't we just want one profile blob by user  id--something like. hmset('userProfileBydId', userId, profile)

File: src/index.js [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+      return this.findUserById(userId);
+    }
+    return null;
+  }
+  async findUserById(userId: string): Promise<?UserObjectType> {
+    const exists = await this.redis.exists(`user:${userId}`);
+    if (!exists) {
+      return null;
+    }
+    const user = await this.redis.hgetall(`user:${userId}`);
```
Comment (grantga) [inline src/index.js]: if the userById has existed you could do a hget('userById', userId)

File: src/index.js [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+    const emails = await this.redis.smembers(`user:${userId}:emails`);
+    let email;
```
Comment (grantga) [inline src/index.js]: Same comment here.

**Comment by TimMikeladze** [comment]
@grantga Good call on using hashes. I made the changes.

**Pull request merged**